src/Raven.Client/Documents/Session/Tokens/IntersectToken.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Text;
+
+namespace Raven.Client.Documents.Session.Tokens
+{
+    public class IntersectToken : QueryToken
```

Comment:
There was no syntax for this in the document,
thought that we might want to support it like this:
`WHERE Name = 'John' INTERSECT Age > 20`.
Thoughts?